### PR TITLE
Add annotation style documentation

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This PR adds some information for advanced styling features that was not documented yet.

They are used for annotations and this documentation is useful in order to standardize, understand and extend the existing local vector styling features. 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

https://mapstore2-offtherailz.readthedocs.io/en/annotation-style-doc/

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: Documentation

### CODE REVIEW

File: docs/developer-guide/maps-configuration.md [hunk 1]
```diff
@@ -728,6 +728,144 @@ The `style` or `styleName` properties of vector layers (wfs, vector...) allow to
...
+              }
+            ]
+          }
+        ]
+      }
+```
+
+*Result:*
+
+<img src="../img/vector-style-annotations.jpg" class="ms-docimage"  style="max-width:600px;"/>
```
Comment (MV88) [inline docs/developer-guide/maps-configuration.md:867]: you sure that this path works when deployed?

File: docs/developer-guide/maps-configuration.md [hunk 2]
```diff
@@ -728,6 +728,144 @@ The `style` or `styleName` properties of vector layers (wfs, vector...) allow to
...
+- a point styled with SVG symbol,
+- a polygon with dashed style
+- a line with start-end point styles as markers with icons
+
+```json
+{
+        "type": "vector",
+        "visibility": true,
+        "id": "1234",
+        "name": "1234",
```
Comment (MV88) [inline docs/developer-guide/maps-configuration.md:778]:
```suggestion
        "id": "styled-vector-layer",
        "name": "styled-vector-layer",
```

File: docs/developer-guide/maps-configuration.md (hunk 1 above)
Comment (offtherailz) [reply docs/developer-guide/maps-configuration.md:867]: it is used here and there in doc. I tested it with mkdocs serve (it is relative so it should be ok). There is some problem you encountered about this ?

Comment (MV88) [reply docs/developer-guide/maps-configuration.md:867]: sounds good

**Pull request merged**